src/main/java/optimus/Parser.java

```diff
@@ -0,0 +1,132 @@
+package optimus;
+
+import optimus.commands.AddTaskCommand;
+import optimus.commands.Command;
+import optimus.commands.DeleteTaskCommand;
+import optimus.commands.LeaveCommand;
+import optimus.commands.ListCommand;
+import optimus.commands.MarkCommand;
+import optimus.commands.UnmarkCommand;
+import optimus.exceptions.IncompleteCommandException;
+import optimus.exceptions.InvalidCommandException;
+import optimus.exceptions.InvalidDateFormatException;
+import optimus.exceptions.InvalidTaskNumberException;
+import optimus.exceptions.OptimusExceptions;
+import optimus.tasks.DeadlineTask;
+import optimus.tasks.EventTask;
+import optimus.tasks.ToDoTask;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class Parser {
+
+    private static final String[] FLAGS = {"/from", "/to", "/by"};
+    private static final String TASK_NOT_SPECIFIED_MSG = "The task number is not specified";
+    private static final String NO_DESCRIPTION_MSG = "This task requires a description";
+
+    /**
+     * Returns a Command object based on user input and throws an Exception for invalid commands
+     * @param input - user input through keyboard
+     * @return corresponding Command object
+     * @throws OptimusExceptions
+     */
+    public static Command parse(String input) throws OptimusExceptions {
+        String[] commands = input.split(" ");
+        if (commands.length == 0) {
+            throw new InvalidCommandException("No command entered.");
+        }
+
+        String command = commands[0].toLowerCase();
+        return switch (command) {
+            case "bye" -> new LeaveCommand();
+            case "list" -> new ListCommand();
+            case "mark" -> new MarkCommand(parseTaskNumber(commands));
+            case "unmark" -> new UnmarkCommand(parseTaskNumber(commands));
+            case "todo", "deadline", "event" -> addTask(commands);
+            case "delete" -> new DeleteTaskCommand(parseTaskNumber(commands));
+            case "find" -> new FindCommand(extractDescription(commands));
+            default -> throw new InvalidCommandException("This command does not exist.");
+        };
+    }
```

Comment:
I believe case and default should not be indented?
```suggestion
        return switch (command) {
        case "bye" -> new LeaveCommand();
        case "list" -> new ListCommand();
        case "mark" -> new MarkCommand(parseTaskNumber(commands));
        case "unmark" -> new UnmarkCommand(parseTaskNumber(commands));
        case "todo", "deadline", "event" -> addTask(commands);
        case "delete" -> new DeleteTaskCommand(parseTaskNumber(commands));
        case "find" -> new FindCommand(extractDescription(commands));
        default -> throw new InvalidCommandException("This command does not exist.");
        };
    }
```